Microsoft.Toolkit.Uwp.SampleApp/SamplePages/CameraPreview/CameraPreviewPage.xaml.cs

```diff
@@ -0,0 +1,147 @@
+                _imageControl.Source = _softwareBitmapSource;
+            }
+
+            _errorMessageText = control.FindDescendantByName("ErrorMessage") as TextBlock;
+
+            semaphoreSlim.Release();
+        }
+
+        protected override void OnNavigatedTo(NavigationEventArgs e)
+        {
+            base.OnNavigatedTo(e);
+            Shell.Current.RegisterNewCommand("Capture Current Frame", CaptureButton_Click);
+            Application.Current.Suspending += Application_Suspending;
+            Application.Current.Resuming += Application_Resuming;
+        }
+
+        protected override void OnNavigatedFrom(NavigationEventArgs e)
+        {
+            CleanUpAsync();
+        }
+
+        private async void Application_Suspending(object sender, SuspendingEventArgs e)
+        {
+            if (Frame.CurrentSourcePageType == typeof(CameraPreviewPage))
+            {
+                var deferral = e.SuspendingOperation.GetDeferral();
+                await CleanUpAsync();
+                deferral.Complete();
+            }
+        }
+
+        private async void Application_Resuming(object sender, object e)
+        {
+            var cameraHelper = new CameraHelper();
+            await _cameraPreviewControl.SetCameraHelperAsync(cameraHelper);
+            _cameraPreviewControl.CameraHelper.FrameArrived += CameraPreviewControl_FrameArrived;
+            _cameraPreviewControl.PreviewFailed += CameraPreviewControl_PreviewFailed;
+        }
+
+        private void CameraPreviewControl_FrameArrived(object sender, FrameEventArgs e)
+        {
+            _currentVideoFrame = e.VideoFrame;
+        }
+
+        private void CameraPreviewControl_PreviewFailed(object sender, PreviewFailedEventArgs e)
+        {
+            _errorMessageText.Text = e.Error;
+        }
+
+        private async void CaptureButton_Click(object sender, RoutedEventArgs e)
+        {
+            var softwareBitmap = _currentVideoFrame.SoftwareBitmap;
+
+            if (softwareBitmap != null)
+            {
+                if (softwareBitmap.BitmapPixelFormat != BitmapPixelFormat.Bgra8 || softwareBitmap.BitmapAlphaMode == BitmapAlphaMode.Straight)
+                {
+                    softwareBitmap = SoftwareBitmap.Convert(softwareBitmap, BitmapPixelFormat.Bgra8, BitmapAlphaMode.Premultiplied);
+                }
+
+                await _softwareBitmapSource.SetBitmapAsync(softwareBitmap);
+            }
+        }
+
+        private async Task CleanUpAsync()
+        {
+            if (_cameraPreviewControl != null)
+            {
+                if (_cameraPreviewControl.CameraHelper != null)
+                {
+                    _cameraPreviewControl.CameraHelper.FrameArrived -= CameraPreviewControl_FrameArrived;
+                }
+
+                _cameraPreviewControl.PreviewFailed -= CameraPreviewControl_PreviewFailed;
+                await _cameraPreviewControl.CleanupAsync();
```

Comment:
I would not expect the control to have a cleanup method, it should react when the CameraHelper is cleaned up and do it on it's own. The CameraHelper should raise an event when the state changes so the control can react appropriately.